scrape/manager_test.go

```diff
@@ -829,25 +904,40 @@ func TestManagerCTZeroIngestion(t *testing.T) {
 						got = append(got, f.f)
 					}
 				}
+
 				if len(app.resultFloats) > 0 {
 					return nil
 				}
 				return fmt.Errorf("expected some samples, got none")
 			}), "after 1 minute")
 			scrapeManager.Stop()
 
-			// Check for zero samples, assuming we only injected always one sample.
-			// Did it contain CT to inject? If yes, was CT zero enabled?
-			if tc.counterSample.CreatedTimestamp.IsValid() && tc.enableCTZeroIngestion {
-				require.Len(t, got, 2)
-				require.Equal(t, 0.0, got[0])
-				require.Equal(t, tc.counterSample.GetValue(), got[1])
-				return
-			}
+			switch tc.typ {
+			case "application/vnd.google.protobuf; proto=io.prometheus.client.MetricFamily; encoding=delimited":
+				// Check for zero samples, assuming we only injected always one sample.
+				// Did it contain CT to inject? If yes, was CT zero enabled?
+				if tc.counterSampleProto.CreatedTimestamp.IsValid() && tc.enableCTZeroIngestion {
+					require.Len(t, got, 2)
+					require.Equal(t, 0.0, got[0])
+					require.Equal(t, tc.counterSampleProto.GetValue(), got[1])
+					return
+				}
+
+				// Expect only one, valid sample.
+				require.Len(t, got, 1)
+				require.Equal(t, tc.counterSampleProto.GetValue(), got[0])
+
+			case "application/openmetrics-text; version=1.0.0; charset=utf-8":
+				require.Len(t, got, len(tc.expectCTLineAppended))
+				for i, e := range tc.expectCTLineAppended {
+					require.Equal(t, e.value, got[i])
+				}
 
-			// Expect only one, valid sample.
-			require.Len(t, got, 1)
-			require.Equal(t, tc.counterSample.GetValue(), got[0])
+				// We expect _created lines to be appended as a new metric if ct ingestion is disabled
+				if !tc.enableCTZeroIngestion {
+					require.Equal(t, "expected_counter_created", app.resultFloats[1].metric.Get(model.MetricNameLabel))
+				}
```

Comment:
The last test case has CT disabled, but no _created line. I'd be impressed if this passes 😬 

I think we'll need en extra boolean for this assertion, we can't rely on `enableCTZeroIngestion`

Reply:
Not sure I follow because the lines highlighted in this comment are specifically for when ct ingestion is disabled. We're only testing if a new _created metric is created here. Maybe I can add comments here explaining this

The rest of the tests are still the same for checking `tc.expectCTLineAppended`, ie we're verifying if metrics with `__name__` == `expected_counter` match the `exp` struct

Reply:
Ah, you're correct, my bad! Reading the test again I believe what you said makes sense :)

Reply:
gonna resolve since approved 🤭